### PR TITLE
feat: email attachment awareness

### LINKED ISSUE #577: feat: email attachment awareness — surface and download email attachments
## Problem

Curia cannot see email attachments. When Nylas returns message data, both the core channel adapter and the ceo-inbox custom skills drop `msg.attachments` during normalization. Agents are completely blind to attached files.

**Symptoms:**
- Curia replies: *"I don't see any attachment in the message — just the text of your email. Could you try resending?"*
- The T2125 expense-tracker references `file-parse` for receipt extraction from attachments but has no mechanism to access them

**Root cause:** `normalizeMessage()` in `src/channels/email/nylas-client.ts` omits `msg.attachments` entirely. The field exists in the Nylas SDK response but is never mapped into `NylasMessage`. Same pattern in `curia-deploy`'s `custom/skills/_lib/nylas-client.ts`.

## Proposed Fix

Three layers of changes across both `curia` and `curia-deploy`:

1. **Surface attachment metadata** — Add `attachments: EmailAttachment[]` (id, filename, contentType, size) to the normalized message types and filter out inline attachments (email signatures, CID images)

2. **Append a human-readable summary** to email body text — `[Attachments: quarterly-report.pdf (245 KB)]` — so agents immediately know about attachments without needing to inspect structured metadata

3. **New download skills** — `email-download-attachment` (core) and `ceo-inbox-download-attachment` (ceo-inbox) that return `content_base64` + `content_type` ready to pipe directly into `file-parse`

## Acceptance Criteria

- [ ] Emailing Curia a file attachment no longer produces "I don't see any attachment"
- [ ] `email-get` response includes `attachments: [{ id, filename, contentType, size }]`
- [ ] `email-list` response includes `attachmentCount` and `hasAttachments` per message
- [ ] `email-download-attachment` skill returns `content_base64` for a valid attachment ID
- [ ] `ceo-inbox-read` includes `attachments` array and `[Attachments: ...]` in `body_plain`
- [ ] `ceo-inbox-download-attachment` skill returns base64 content
- [ ] Inline attachments (email signatures, CID images) are filtered out
- [ ] Attachments ≥ 10 MB are rejected with a clear error
- [ ] All new and modified handlers have test coverage
- [ ] TypeScript types pass without errors in both repos

## Scope

**`curia` repo:** `nylas-client.ts`, `message-converter.ts`, `outbound-gateway.ts`, `email-get`, `email-list`, new `email-download-attachment` skill, `coordinator.yaml`

**`curia-deploy` repo:** `_lib/nylas-client.ts`, `ceo-inbox-read`, `ceo-inbox-list`, new `ceo-inbox-download-attachment` skill, `ceo-inbox.yaml`, `T2125-expense-tracker.yaml`

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- **`NylasMessage`** now carries a typed `attachments: EmailAttachment[]` field; `normalizeMessage()` filters inline parts (embedded images, CID graphics) using both `isInline` and `contentDisposition.startsWith('inline')`
- **Email body** gets a human-readable `[Attachments: report.pdf (245 KB)]` summary appended so agents see attachments without any prompt changes
- **`email-get`** returns full attachment metadata per message; **`email-list`** returns `attachmentCount`/`hasAttachments` flags
- **New `email-download-attachment` skill** — verifies the attachment exists, enforces a 10 MB cap (pre- and post-download), and returns `content_base64` + `content_type` ready to pipe directly into `file-parse`
- **`OutboundGateway`** gains `downloadEmailAttachment()` using the same named-client routing as all other email methods
- Coordinator agent updated with the three-step attachment workflow

Closes #577

## Test plan

- [ ] 2266 unit tests passing (run `pnpm test`)
- [ ] Send an email with a file attachment to Curia's address — the inbound message event should include `metadata.attachments` and the content should end with `[Attachments: ...]`
- [ ] Call `email-get` on the message — response includes `attachments` array with id, filename, contentType, size
- [ ] Call `email-download-attachment` with the attachment id + message id — returns base64 content
- [ ] Pipe `content_base64` + `content_type` to `file-parse` — parses successfully
- [ ] Attachment ≥ 10 MB returns a clear error (pre-check on declared size; post-check catches size=0 bypass)
- [ ] Inline attachments (email signature images) are filtered out and not surfaced to agents

## Release Notes

**New Features**
- Download email attachments directly from messages
- View attachment counts and availability flags in email lists
- Attachment metadata and summaries now included in email content

**Documentation**
- Updated guidance for email attachment handling workflow

[Review Change Stack](https://app.coderabbit.ai/change-stack/josephfung/curia/pull/579)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ bus event types) are noted explicitly even in the `0.x` range.
 
 ### Added
 
+- **Email attachment support** — `NylasMessage` now carries a typed `attachments` array; the message converter appends a human-readable attachment summary to email content; `email-get` returns full attachment metadata and `email-list` returns `attachmentCount`/`hasAttachments` flags; new `email-download-attachment` skill downloads attachment bytes as base64 ready for `file-parse`.
```
[the remark this answers is not in the document]
Comment (josephfung) [reply CHANGELOG.md]: Fixed — condensed to a single clause under 15 words. New text: *"surface metadata in email skills, append human-readable summaries to message content, and download bytes as base64 for `file-parse`."*

File: src/skills/outbound-gateway.ts [hunk 2]
```diff
@@ -959,6 +959,26 @@ export class OutboundGateway {
...
+   */
+  async downloadEmailAttachment(
+    attachmentId: string,
+    messageId: string,
+    accountId?: string,
+  ): Promise<Buffer> {
+    const client = this.getNylasClient(accountId);
+    if (!client) {
+      throw new Error('outbound-gateway: downloadEmailAttachment called but no nylasClient is configured');
+    }
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/skills/outbound-gateway.ts:978]: Declining this change. In this codebase, `AgentError` is a plain object interface (not a throwable class) — it's created by `classifyError()` and `classifySkillError()` in the error recovery path, not thrown directly. The existing pattern throughout `outbound-gateway.ts` is `throw new Error()` for misconfiguration (lines 957, 992, 1282, 1343, 1357 all do the same thing). The skill handler already catches this and returns `{ success: false, error: ... }`. Changing only this one method in isolation would create inconsistency without making the error handling more correct.

**Comment by josephfung** [comment]
**Re: nitpick on `message-converter.test.ts` (lines 404–459):** Added 3 edge-case tests in the same commit as the CHANGELOG fix:
- **Positioning** — asserts the attachment summary appears *after* the body content (not merely present somewhere in `content`)
- **Empty body coexistence** — asserts `"(empty email)"` and `[Attachments: ...]` both appear in the correct order when body and snippet are both empty
- **Zero-size** — asserts a zero-byte attachment formats as `"0 B"`
